### DISCUSSION TITLE
Using `tag="!!null"` on a whole document to make nulls blank

### DISCUSSION THREAD

**Discussion by little-flag** [mikefarah/yq#2092] [General]
I want to take this:

```
title: A title
author: ""
pages: null
```

And get the output:

```
title: A title
author: 
pages: 
```

I found [this example](https://stackoverflow.com/a/67072611):

`yq eval --null-input '.a="" | .a tag="!!null"'` 

But I don't understand how to do this for a whole document.

Many thanks in advance for any advice.

**Comment by mikefarah** [comment]
You can do it like this:

```
yq 'with(.. | select(tag == "!!null" or . == ""); . = "" | . tag="!!null")' data1.yaml
```

Explanation:
- `.. | select(tag == "!!null" or . == "")` - this matches all the nodes in the tree, and selects the ones that are either null or the empty string
- `with(<exp>; blah)` - this is a convenient way to make multiple updates to each match ([doc](https://mikefarah.gitbook.io/yq/operators/with))
- Then, within the `with` loop if you like, you update each element as you had.

**Comment by little-flag** [comment]
Thank you for the code and explanation but this is only working for me with the nulls, not the empty strings.

This: 

`echo "title: A title\nauthor: \"\"\npages: null" | yq 'with(.. | select(tag == "!!null" or . == ""); . = "" | . tag="!!null")'`

Produces this:

```
title: A title
author: ""
pages:
```

**Comment by Rack34** [comment]
Mn
